block/src/Pos/Block/Logic/Internal.hs

```diff
@@ -231,40 +236,40 @@ rollbackBlocksUnsafe bsc scb toRollback = do
     sanityCheckDB
 
 
-toComponentBlock :: HasConfiguration => (MainBlock -> payload) -> Block -> ComponentBlock payload
+toComponentBlock :: (MainBlock -> payload) -> Block -> ComponentBlock payload
 toComponentBlock fnc block = case block of
     Left genBlock   -> ComponentBlockGenesis (convertGenesis genBlock)
     Right mainBlock -> ComponentBlockMain (Some $ mainBlock ^. gbHeader) (fnc mainBlock)
 
 toTxpBlock
-    :: HasConfiguration
+    :: ()
     => Block -> TxpBlock
```

Comment:
You should probably drop the constraint on that all together so it becomes:
```
toTxpBlock
    :: Block -> TxpBlock
```
and similar for the rest.